amaranth/back/rtlil.py

```diff
@@ -859,37 +859,68 @@ def emit_read_port(self, cell_idx, cell):
             })
         self.builder.cell(f"$memrd_v2", ports=ports, params=params, src=_src(cell.src_loc))
 
-    def emit_property(self, cell_idx, cell):
-        if isinstance(cell, _nir.AsyncProperty):
-            ports = {
-                "A": self.sigspec(cell.test),
-                "EN": self.sigspec(cell.en),
-            }
-        if isinstance(cell, _nir.SyncProperty):
-            test = self.builder.wire(1, attrs={"init": _ast.Const(0, 1)})
-            en = self.builder.wire(1, attrs={"init": _ast.Const(0, 1)})
-            for (d, q) in [
-                (cell.test, test),
-                (cell.en, en),
-            ]:
-                ports = {
-                    "D": self.sigspec(d),
-                    "Q": q,
-                    "CLK": self.sigspec(cell.clk),
-                }
-                params = {
-                    "WIDTH": 1,
-                    "CLK_POLARITY": {
-                        "pos": True,
-                        "neg": False,
-                    }[cell.clk_edge],
-                }
-                self.builder.cell(f"$dff", ports=ports, params=params, src=_src(cell.src_loc))
-            ports = {
-                "A": test,
-                "EN": en,
-            }
-        self.builder.cell(f"${cell.kind}", name=cell.name, ports=ports, src=_src(cell.src_loc))
+    def emit_print(self, cell_idx, cell):
+        args = []
+        format = []
+        if cell.format is not None:
+            for chunk in cell.format.chunks:
+                if isinstance(chunk, str):
+                    format.append(chunk)
+                else:
+                    args += chunk.value
```

Comment:
this is wrong for the `s` type, as they need byteswapping to match Verilog string semantics